client.go

```diff
@@ -50,6 +51,14 @@ type PublishResult struct {
 	Offset    uint64
 }
 
+// EventHandler is a function to process the messages read from the stream and is passed as
+// a parameter to the subscribe call.
+type EventHandler = func(ctx context.Context, payload []byte, contentType string, headers map[string]string) error
+
+// EventErrHandler is a function to handle errors while reading subscription messages and
+// is passed as a parameter to the subscribe call.
+type EventErrHandler = func(cancel context.CancelFunc, err error)
```

Comment:
Why does the error handler get the cancelFunc? There are cases below when the error handler gets triggered **because** the context got (already) canceled.

Reply:
The error handler can choose to cancel the subscription by calling the cancelFunc if it chooses to. I will update the doc.

It is ok to call the cancelFunc multiple times since it then becomes a no-op.